x-hooks/use-bounty.tsx

```diff
@@ -0,0 +1,203 @@
+  changeCurrentBountyData,
+  changeCurrentBountyDataChain,
+  changeCurrentBountyDataIsDraft,
+  changeCurrentBountyDataIsFinished,
+  changeCurrentBountyDataIsInValidation,
+} from "../contexts/reducers/change-current-bounty";
+import {bountyReadyPRsHasNoInvalidProposals} from "../helpers/proposal";
+import {IssueData} from "../interfaces/issue-data";
+import useApi from "./use-api";
+import useOctokit from "./use-octokit";
+import {changeSpinners} from "../contexts/reducers/change-spinners";
+
+
+const CACHE_BOUNTY_TIME = 60 * 1000; // 1min
+
+export const BountyContext = createContext({});
+export const BountyProvider = ({children}) => <BountyContext.Provider value={null} children={children} />
+
+export function useBounty() {
+  if (!useContext(BountyContext))
+    throw new Error(`useBounty not inside BountyContext`);
+
+  const {state, dispatch} = useAppState();
+
+  const {query} = useRouter();
+  const {getIssue} = useApi();
+  const { getIssueOrPullRequestComments, getPullRequestDetails } = useOctokit();
+
+  function isCurrentBountyCached() {
+    const lastUpdated = state.currentBounty?.lastUpdated;
+    const {id, githubId, repository_id} = state.currentBounty?.data || {};
+
+    if (id && query.id === githubId && +query.repoId === repository_id)
+      if (lastUpdated && +new Date() - lastUpdated <= CACHE_BOUNTY_TIME)
+        return true;
+
+    return false;
+  }
+
+  function getDatabaseBounty(force = false) {
+    if (!state.Service?.network?.active || (!query?.id || !query.repoId))
+      return;
+
+    console.debug(`Loading bounty information`, state.spinners?.bountyDatabase);
+
+    if (!force && isCurrentBountyCached() || state.spinners?.bountyDatabase)
+      return;
+
+    console.debug(`GET ISSUE`, state.Service)
+
+    dispatch(changeSpinners.update({bountyDatabase: true}))
+
+    getIssue(+query.repoId, +query.id, state.Service.network.active.name)
+      .then((bounty: IssueData) => {
+
+        console.debug(`GOT ISSUE`);
+
+        const bigNumbers = {
+          amount: BigNumber(bounty.amount),
+          fundingAmount: BigNumber(bounty.fundingAmount),
+          fundedAmount: BigNumber(bounty.fundedAmount)
+        }
+
+        const mergeProposalMapper = (proposal) => ({
+          ...proposal,
+          isMerged: bounty.merged !== null && proposal.scMergeId === bounty.merged
+        })
+
+        const mergeProposals = bounty.mergeProposals.map(mergeProposalMapper);
+        dispatch(changeCurrentBountyData({...bounty, mergeProposals, ...bigNumbers}));
+        return getIssueOrPullRequestComments(bounty.repository.githubPath, +bounty.githubId);
+      })
+      .then(comments => {
+        dispatch(changeCurrentBountyComments(comments));
+        dispatch(changeSpinners.update({bountyDatabase: false}))
+      });
+
+  }
+
+  function getChainBounty(force = false) {
+    if (!state.Service?.active || !state.Service?.network || !state.currentBounty?.data?.contractId || state.spinners?.bountyChain)
```

Comment:
`!state.currentBounty?.data?.contractId` can't be used because `contractId` of a bounty can be `0`

```suggestion
    if (!state.Service?.active || !state.Service?.network || state.currentBounty?.data?.contractId === null || state.currentBounty?.data?.contractId === undefined || state.spinners?.bountyChain)
```

Reply:
Is a bit weird that `contractId` can be 0 since the `0` bounty on the contract is an empty bounty :O